### PR TITLE
coveralls

### PR DESCRIPTION
**Pull request by weedySeaDragon**
PT Story: **Set up coveralls**
https://www.pivotaltracker.com/story/show/133925105

Changes proposed in this pull request:
1.  add the coverall badge to the Readme
2. use coveralls in the setup for both features and rspec tests
3. don't commit coveralls reports into the repos ( = ignore /coverage in .gitignore)

Questions: 
- do we need to put the repo token into `.coveralls.yml`?
- do we need a lib/tasks/ci.rake like the one in [WebsiteOne](https://github.com/diraulo/WebsiteOne/blob/d908be95e847743e5c9319fa447869018dd4e1d7/lib/tasks/ci.rake)?

Ready for review:
@diraulo @tochman

### CODE REVIEW

**Comment by diraulo** [comment]
@weedySeaDragon 

> do we need to put the repo token into .coveralls.yml?

No. The `repo_token` will be used directly on semaphore when executing the build command. eg `COVERALLS_REPO_TOKEN=paste_token_from_coveralls bundle exec rake`

> do we need a lib/tasks/ci.rake like the one in WebsiteOne?

Yes. The following content should be enough for that

``` ruby
unless Rails.env.production?
  require 'rspec/core/rake_task'
  require 'cucumber/rake/task'
  require 'coveralls/rake/task'

  Coveralls::RakeTask.new

  namespace :ci do
    desc 'Run Rspec and Cucumber then push coverage report to coveralls'
    task tests: [:spec, :cucumber, 'coveralls:push']
  end
end
```

The above code will create a tasks that'll run both RSpec and Cucumber, then push the coverage report to coveralls.

File: features/support/env.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 require 'cucumber/rails'
+require 'coveralls'
+
+Coveralls.wear_merged!('rails')
```
Comment (diraulo) [inline features/support/env.rb:2]: It's recommended to have both coveralls lines at the top of the file, before anything else.

**Review by diraulo — APPROVED** [review]
As soon as we merge this. Someone needs to update the build on semaphore and replace:

``` shell
bundle exec rspec
bundle exec cucumber
```

with

``` shell
COVERALLS_REPO_TOKEN=paste_token_from_coveralls bundle exec rails ci:tests
```

**Comment by weedySeaDragon** [comment]
I can't see how to change the settings for the Semaphore build (to now call `... rails ci:tests`).  Is this because the project belongs to @lollypop27 on Semaphore?  Or am I just not looking in the right place for the settings?

**Comment by tochman** [comment]
I have access to those settings. Is this ready to be merged?

**Comment by weedySeaDragon** [comment]
I've made the changes @diraulo suggested, so if the semaphore change is the only thing left to do, then yes,  it's ready to be merged.

**Pull request merged**